libs/ml/tests/ml/dataloader/tensor_dataloader.cpp

```diff
@@ -60,6 +60,8 @@ TYPED_TEST(TensorDataloaderTest, serialize_tensor_dataloader)
   // initialise a new tensor dataloader with the wrong shape parameters
   // these will get updated by deserialisation
   fetch::ml::dataloaders::TensorDataLoader<TypeParam, TypeParam> tdl_2({1, 1}, {{1, 1}});
+  tdl_2.SetTestRatio(0.5);
```

Comment:
what happens if the dataset is a single example?
Does it have a validation and test size of 0 as intended?

maybe we should cover this with a test?